wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/impl/FactoryEditValidatorImpl.java

```diff
@@ -41,13 +41,17 @@
      */
     @Override
     public void validate(Factory factory) throws ForbiddenException, ServerException {
-        // ensure user has the correct permissions
-        final String userId = EnvironmentContext.getCurrent().getSubject().getUserId();
         // Checks if there is an author from the factory (It may be missing for some old factories)
         Author author = factory.getCreator();
         if (author == null || author.getUserId() == null) {
             throw new ServerException(format("Invalid factory without author stored. Please contact the support about the factory ID '%s'",
                                              factory.getId()));
         }
+        // ensure user has the correct permissions
+        final String userId = EnvironmentContext.getCurrent().getSubject().getUserId();
+        if (userId == null || !userId.equals(author.getUserId())) {
```

Comment:
as author is not null and author.getUserId() not null due to previous block check
it can be simplified to 
if (!author.getUserId().equals(userId)) {